src/smpackage/ZipArchive/stl/ZipString.h

```diff
@@ -142,6 +142,7 @@ class ZIP_API CZipString : public stdbs
 			int nLen = sizeof(TCHAR) * iCounter*1024;
 			pBuf = (TCHAR*)realloc((void*)pBuf, nLen);
 			if (!pBuf)
+				va_end (arguments);
```

Comment:
Adding this line is not valid because it changes the if statement.

The old code is equivalent to this:
```c++
if(!pBuf)
{
  return;
}
```
The new code is equivalent to this:
```c++
if(!pBuf)
{
  va_end(arguments);
}
return;
```